File: src/utils/logger.js
const clc = require('cli-color');

const log = {
	error: (msg) => console.log(clc.red.bold('Error: ') + msg),
	warn: (msg) => console.log(clc.yellow('Warn: ') + msg),
	info: (msg) => console.log(clc.blue('Info: ') + msg),
	success: (msg) => console.log(clc.green('Success: ') + msg),
	debug: (msg) => process.env.DEBUG && console.log(clc.magenta('Debug: ') + msg),
};

module.exports = log;

File: src/cli/logger.cpp
#include <iostream>

#include "logger.h"
#include "styles.h"

namespace logger {
    void logInfo(const std::string& message) {
        std::cout << styles::blue("INFO: ") << message << std::endl;
    }

    void logWarning(const std::string& message){ 
        std::cout << styles::yellow("WARNING: ") << message << std::endl;
    }

    void logError(const std::string& message) {
        std::cerr << styles::red("ERROR: ") << message << std::endl;
    }

    void logDebug(const std::string& message) {
        std::cout << styles::magenta("DEBUG: ") << message << std::endl;
    }

    void logSuccess(const std::string& message) { 
        std::cout << styles::green("SUCCESS: ") << message << std::endl;
    }
}
